### PR TITLE
Don't repeat the Oracle password in ci/build.yml

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
Got it from https://stackoverflow.com/a/13802438

Incidentally, to write this commit message containing a single quote, I had to escape it with this https://stackoverflow.com/a/1250279

I guess today is "bring-your-quotation-marks-to-work day".

changelog_begin
changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ci/build.yml [hunk 1]
```diff
@@ -230,7 +230,7 @@ jobs:
         }
         trap cleanup EXIT
         testConnection() {
-            docker exec oracle bash -c 'sqlplus -L system/hunter2@//localhost:1521/ORCLPDB1 <<< "select * from dba_users;"; exit $?' >/dev/null
+            docker exec oracle bash -c 'sqlplus -L system/'"$ORACLE_PWD"'@//localhost:1521/ORCLPDB1 <<< "select * from dba_users;"; exit $?' >/dev/null
```
Comment (cocreature) [inline ci/build.yml:233]: We could also use double quotes for the outer one and then single quotes for the `select` and escape the `$?`. Not sure that’s clearer, wdyt?

**Review by cocreature — APPROVED** [review]
*******

File: ci/build.yml (hunk 1 above)
Comment (garyverhaegen-da) [reply ci/build.yml:233]:
```suggestion
            docker exec oracle bash -c <<BASH
                sqlplus -L system/$ORACLE_PWD@//localhost:1521/ORCLPDB1 <<< "select * from dba_users;" >/dev/null
            BASH
```

I think this should also work, though it's arguably a slightly different semantics:

```
docker exec oracle bash -c "sqlplus -L system/$ORACLE_PWD@//localhost:1521/ORCLPDB1" <<< "select * from dba_users;" >/dev/null
```

**Comment by garyverhaegen-da** [comment]
Note that the branch name is misleading: today you learned that _there is no way to expand a variable within single quotes_ and that the workaround is to close the single quotes before expanding the variable.

File: ci/build.yml (hunk 1 above)
Comment (cocreature) [reply ci/build.yml:233]: I think the `exit` stuff was required but I forgot why :thinking:

Comment (garyverhaegen-da) [reply ci/build.yml:233]: Bash returns the exit code of the last command by default AFAIK so I don't think it does anything here. Maybe the command used to be more complex?

Comment (cocreature) [reply ci/build.yml:233]: I thought it was some mix of `docker exec` and `bash -c` that required it but I’ve probably just messed it up at the time.

Comment (stefanobaghino-da) [reply ci/build.yml:233]: I don't have any specific opinion, but this works and seems fairly readable.

**Comment by stefanobaghino-da** [comment]
> Note that the branch name is misleading: today you learned that _there is no way to expand a variable within single quotes_ and that the workaround is to close the single quotes before expanding the variable.

😂 

Indeed.

😂

**Pull request merged**